internal/dag/status_test.go

```diff
@@ -5545,6 +5546,193 @@ func TestGatewayAPIHTTPRouteDAGStatus(t *testing.T) {
+			&gatewayapi_v1.HTTPRoute{
+				TypeMeta: meta_v1.TypeMeta{
+					Kind: KindHTTPRoute,
+				},
+				ObjectMeta: meta_v1.ObjectMeta{
+					Name:              "basic-3",
+					Namespace:         "default",
+					CreationTimestamp: meta_v1.NewTime(time.Date(2022, time.Month(2), 21, 1, 10, 30, 0, time.UTC)),
+				},
+				Spec: gatewayapi_v1.HTTPRouteSpec{
+					CommonRouteSpec: gatewayapi_v1.CommonRouteSpec{
+						ParentRefs: []gatewayapi_v1.ParentReference{gatewayapi.GatewayParentRef("projectcontour", "contour")},
+					},
+					Hostnames: []gatewayapi_v1.Hostname{
+						"test.projectcontour.io",
+					},
+					Rules: []gatewayapi_v1.HTTPRouteRule{{
+						Matches: []gatewayapi_v1.HTTPRouteMatch{
+							{
+								Path: &gatewayapi_v1.HTTPPathMatch{
+									Type:  ptr.To(gatewayapi_v1.PathMatchPathPrefix),
+									Value: ptr.To("/"),
+								},
+								Headers: []gatewayapi_v1.HTTPHeaderMatch{
+									{
+										Type:  ptr.To(gatewayapi_v1.HeaderMatchExact),
+										Name:  gatewayapi_v1.HTTPHeaderName("foo"),
+										Value: "bar",
+									},
+								},
+								QueryParams: []gatewayapi_v1.HTTPQueryParamMatch{
+									{
+										Type:  ptr.To(gatewayapi_v1.QueryParamMatchRegularExpression),
+										Name:  "param-1",
+										Value: "valid-[a-z]?-[A-Za-z]+-[0=9]+-\\d+",
+									},
+								},
+							},
+						},
+
+						BackendRefs: gatewayapi.HTTPBackendRef("kuard", 8080, 1),
+					}},
+				},
+			},
+		},
+		wantRouteConditions: []*status.RouteStatusUpdate{
+			{
+				FullName: types.NamespacedName{Namespace: "default", Name: "basic-1"},
+				RouteParentStatuses: []*gatewayapi_v1.RouteParentStatus{
+					{
+						ParentRef: gatewayapi.GatewayParentRef("projectcontour", "contour"),
+						Conditions: []meta_v1.Condition{
+							routeResolvedRefsCondition(),
+							routeAcceptedHTTPRouteCondition(),
+						},
+					},
+				},
+			},
+			{
+				FullName: types.NamespacedName{Namespace: "default", Name: "basic-2"},
+				RouteParentStatuses: []*gatewayapi_v1.RouteParentStatus{
+					{
+						ParentRef: gatewayapi.GatewayParentRef("projectcontour", "contour"),
+						Conditions: []meta_v1.Condition{
+							routeAcceptedFalse(status.ReasonRouteConflict, "HTTPRoute's Match has conflict with other HTTPRoute's Match"),
+							routeResolvedRefsCondition(),
+						},
+					},
+				},
+			},
+			{
+				FullName: types.NamespacedName{Namespace: "default", Name: "basic-3"},
+				RouteParentStatuses: []*gatewayapi_v1.RouteParentStatus{
+					{
+						ParentRef: gatewayapi.GatewayParentRef("projectcontour", "contour"),
+						Conditions: []meta_v1.Condition{
+							routeAcceptedFalse(status.ReasonRouteConflict, "HTTPRoute's Match has conflict with other HTTPRoute's Match"),
+							routeResolvedRefsCondition(),
+						},
+					},
+				},
+			},
+		},
+		// is it ok to show the listeners are attached, just it's not accepted because of the conflict
```

Comment:
My question here, the conflicted route is still shown attached. Is that ok?

Reply:
Yep, that is correct:

```
	// Successful attachment of a Route to a Listener is based solely on the
	// combination of the AllowedRoutes field on the corresponding Listener
	// and the Route's ParentRefs field. A Route is successfully attached to
	// a Listener when it is selected by the Listener's AllowedRoutes field
	// AND the Route has a valid ParentRef selecting the whole Gateway
	// resource or a specific Listener as a parent resource (more detail on
	// attachment semantics can be found in the documentation on the various
	// Route kinds ParentRefs fields). Listener or Route status does not impact
	// successful attachment, i.e. the AttachedRoutes field count MUST be set
	// for Listeners with condition Accepted: false and MUST count successfully
	// attached Routes that may themselves have Accepted: false conditions.
```

From: https://github.com/kubernetes-sigs/gateway-api/blob/72f679c7c67228f5b4340e0c2f50cc04b36945b1/apis/v1/gateway_types.go#L849-L859